### DISCUSSION TITLE
Ban explicit use of unknown type

### DISCUSSION THREAD

**Discussion by jpike88** [rome/tools#4332] [Suggestions]
When Rome is capable of reading types...

Casting to unknown type is an escape hatch to allow you to 'force' one type to act as another.

It's just too dangerous from our point of view to allow for it, there should be a rule that just prevents using it altogether.

e.g.

```
// ban this
const x: unknown = '';

// ban this
const x: number = 3;
const y: string = x as unknown as string;

```

**Comment by Conaclos** [comment]
I don't see your point for the first example. `unknown` is "safe" in comparison to `any`. It could be a pretty bad experience to disallow this.

The second one is more arguable. In some cases it represents a required escape hatch because the type could be too hard to express in _TypeScript_. By the way this could be a good thing to require an explicit comment to explain why we need that.

We would like to have more feedbacks from other devs, and possibly if the usage of this pattern is frequent.

**Reply by jpike88** [reply]
Yes this is a bad idea, after auditing our code a little more I think we could use it.
